sql/parser/builtins.go

```diff
@@ -761,3 +773,73 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+// Round rounds x to n digits.
+// Directly taken from CPython's `double_round` in
+// cpython/Object/floatobject.c.
+//
+// TODO(thschroeter):
+// - understand how it works
+// - bounds check for n
+// - test overflow
+func roundDigits(x float64, n int64) float64 {
+	var pow1, pow2, y, z float64
+	ndigits := float64(n)
+
+	if ndigits >= 0 {
+		if ndigits > 22 {
```

Comment:
But I'm not sure I understand what such a large number of digits would even mean since doubles don't have 22 decimal digits of precision.